Mathlib/Topology/Separation.lean

```diff
@@ -2286,51 +2354,40 @@ instance (priority := 100) NormalSpace.of_compactSpace_r1Space [CompactSpace X]
     NormalSpace X where
   normal _s _t hs ht := .of_isCompact_isCompact_isClosed hs.isCompact ht.isCompact ht
 
-/-- A regular topological space with second countable topology is a normal space.
+/-- A regular topological space with a Lindelöf topology is a normal space. A consequence of e.g.
+Corollaries 20.8 and 20.10 of Willard's *General Topology* <https://zbmath.org/1052.54001> (without
+the assumption of Hausdorff). -/
+instance (priority := 100) NormalSpace.of_regularSpace_lindelofSpace
+    [r: RegularSpace X] [LindelofSpace X] : NormalSpace X where
+  normal h k hcl kcl hkdis := by
+    wlog nonempty_space : Nonempty X
+    · rw [not_nonempty_iff] at nonempty_space
+      exact normal h k hcl kcl hkdis
+    have disjoint_cover_lemma (h₀ k₀ : Set X) (k₀cl : IsClosed k₀) (h₀k₀dis : Disjoint h₀ k₀)
+        (a : X) : ∃ n : Set X, IsOpen n ∧ Disjoint (closure n) k₀ ∧ (a ∈ h₀ → a ∈ n) := by
+      wlog ainh₀ : a ∈ h₀
+      · exact ⟨∅, isOpen_empty, SeparatedNhds.disjoint_closure_left (SeparatedNhds.empty_left k₀),
+          fun a ↦ ainh₀ a⟩
+      obtain ⟨n, nna, ncl, nsubkc⟩ := (((regularSpace_TFAE X).out 0 3).mp r:) a k₀ᶜ
+        (k₀cl.compl_mem_nhds (disjoint_left.mp h₀k₀dis ainh₀))
+      exact ⟨interior n, isOpen_interior, disjoint_left.mpr fun ⦃a⦄ ain ↦
+        nsubkc ((IsClosed.closure_subset_iff ncl).mpr interior_subset ain),
+        fun _ ↦ mem_interior_iff_mem_nhds.mpr nna⟩
+    choose u u_open u_dis u_nhd using disjoint_cover_lemma h k kcl hkdis
```

Comment:
I am still thinking about a way to avoid some code duplication here.

Reply:
I think 7286de7 is what we want.